src/cloud_provider/utilities.rs

```diff
@@ -242,33 +244,103 @@ fn get_version_number(version: &str) -> Result<VersionsNumber, StringError> {
     Ok(VersionsNumber { major, minor, patch })
 }
 
+fn cloudflare_dns_resolver() -> Resolver {
+    let mut resolver_options = ResolverOpts::default();
+
+    //  We want to avoid cache and using hostfile of the host, as some provider force caching
+    //  which lead to stale response
+    resolver_options.cache_size = 0;
+    resolver_options.use_hosts_file = false;
+
+    Resolver::new(ResolverConfig::cloudflare(), resolver_options)
+        .expect("Invalid cloudflare DNS resolver configuration")
+}
+
+fn get_domain_under_cname(resolver: &Resolver, cname: &str) -> Option<String> {
```

Comment:
no clear, I suggest to rename to get_cname_record_value

Reply:
done